File: LeetCode_3202/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maximumLength(vector<int>& nums, int k) {
        int res = 2;
        vector<vector<int>> dp(k, vector<int> (k, 0));

        for (int num : nums) {
            num %= k;
            for (int prev = 0; prev < k; prev++) {
                dp[num][prev] = dp[prev][num] + 1;
                res = max(res, dp[num][prev]);
            }
        }
        return res;
    }
};

File: LeetCode_3202/Main.js

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximumLength = function(nums, k) {
    const dp = Array.from({ length: k }, () => new Array(k).fill(0));
    for (let i = 0; i < k; i++) {
        for (let j = 0; j < k; j++) {
            dp[i][j] = 0;
        }
    }
    let res = 2;
    for (let num of nums) {
        num %= k;
        for (let prev = 0; prev < k; prev++) {
            dp[prev][num] = dp[num][prev] + 1;
            res = Math.max(res, dp[prev][num]);
        }
    }
    return res;
};